internal/manager/run.go

```diff
@@ -62,8 +61,22 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
 		return fmt.Errorf("unable to build the kong admin api configuration: %w", err)
 	}
 
+	kongRoot, err := kongConfig.Client.Root(ctx)
+	if err != nil {
+		return fmt.Errorf("could not retrieve Kong admin root: %w", err)
+	}
+	kongRootConfig, ok := kongRoot["configuration"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("invalid root configuration, expected a map[string]interface{} got %T",
+			kongRoot["configuration"])
+	}
+	dbmode, ok := kongRootConfig["database"].(string)
+	if !ok {
+		return fmt.Errorf("invalid database configuration, expected a string got %T", kongRootConfig["database"])
+	}
+
```

Comment:
Is there some way we could expose a generic "get arbitrary JSON path from the root output" (or any output, I guess)? That'd simplify this general pattern a lot, and avoid a lot of similar functions (since the root output is quite large).

I didn't DRY here since run.go is the abnormal area of code we don't have access to this already from the proxy object (sort of--we don't have a `GetDBMode()` but could add one easily)

Reply:
> Is there some way we could expose a generic "get arbitrary JSON path from the root output" (or any output, I guess)? That'd simplify this general pattern a lot, and avoid a lot of similar functions (since the root output is quite large).

I've used https://github.com/tidwall/gjson in the past, however using this as opposed to having JSON annotated structs in our codebases feels _a little bit wrong_ if nothing else because of the precedent being strong in our codebases for the annotation method of actually defining the structure.